libs/SalesforceSDK/src/com/salesforce/androidsdk/push/SFDCFcmListenerService.java

```diff
@@ -26,30 +26,28 @@
  */
 package com.salesforce.androidsdk.push;
 
-import android.os.Bundle;
-
-import com.google.android.gms.gcm.GcmListenerService;
+import com.google.firebase.messaging.FirebaseMessagingService;
+import com.google.firebase.messaging.RemoteMessage;
 import com.salesforce.androidsdk.app.SalesforceSDKManager;
 
-public class SFDCGcmListenerService extends GcmListenerService {
-
-    public static final String KEY_FROM = "from";
+/**
+ * This class is called when a message is received or the token changes.
+ *
+ * @author bhariharan
+ */
+public class SFDCFcmListenerService extends FirebaseMessagingService {
 
     /**
      * Called when message is received.
      *
-     * @param from SenderID of the sender.
-     * @param data Data bundle containing message data as key/value pairs.
+     * @param message Remote message received.
      */
     @Override
-    public void onMessageReceived(String from, Bundle data) {
-        if (data != null && SalesforceSDKManager.hasInstance()) {
+    public void onMessageReceived(RemoteMessage message) {
+        if (message != null && SalesforceSDKManager.hasInstance()) {
             final PushNotificationInterface pnInterface = SalesforceSDKManager.getInstance().getPushNotificationReceiver();
             if (pnInterface != null) {
-
-                // Add 'from' to the Bundle directly before passing it back.
-                data.putString(KEY_FROM, from);
```

Comment:
We don't need this anymore because `RemoteMessage` has a dedicated field for sender, i.e. `from` field.